### PR TITLE
Remove static pool from AllocationMap

### PR DESCRIPTION
**Pull request by jdahm**

### CODE REVIEW

**Comment by jdahm** [comment]
I tested that this fixes the extended static initialization tests on the LC systems.

File: src/umpire/strategy/DynamicPool.cpp [hunk 1]
```diff
@@ -277,6 +277,8 @@ void DynamicPool::mergeFreeBlocks()
 {
   using PointerMap = std::map<Pointer, SizeTuple>;
 
+  if (m_free_map.size() < 2) return;
```
Comment (jdahm) [inline src/umpire/strategy/DynamicPool.cpp]: This is unrelated, but it's good to return here rather than build a map unnecessarily.

Comment (mcfadden8) [reply src/umpire/strategy/DynamicPool.cpp]: This looks fine, although I would be inclined to move this check above the `using Pointermap` declaration on line 278.  I realize it won't make any runtime difference and it is a cosmetic/style thing.  So, if you want to keep the line here, that is fine too.

File: src/umpire/strategy/DynamicPool.cpp [hunk 2]
```diff
@@ -51,23 +51,13 @@ DynamicPool::~DynamicPool()
...
-  }
+  // Warning if blocks are still in use
   if (m_used_map.size() > 0) {
     std::stringstream ss;
-    ss << "Pointers still in use at destruction time.";
-    if (num_heads > 0) {
-      ss << " This will cause " << num_heads << " leak(s).";
-    }
-    UMPIRE_LOG(Error, ss.str());
+    ss << "The following addresses have not been deallocated:";
```
Comment (davidbeckingsale) [inline src/umpire/strategy/DynamicPool.cpp]: I think you should maybe check the size of the map, because you don't want to print a million leaks...

**Pull request merged**